src/index.js

```diff
@@ -0,0 +1,164 @@
+const OLD_EXEC_PATH_CONFIG_KEY = 'linter-rubocop.executablePath';
+const OLD_ARGS_CONFIG_KEY = 'linter-rubocop.additionalArguments';
+const DEFAULT_LOCATION = { line: 1, column: 1, length: 0 };
+const DEFAULT_ARGS = [
+  '--cache', 'false',
+  '--force-exclusion',
+  '--format', 'json',
+  '--display-style-guide',
+];
+const AUTOCORRECT_ARG = '--auto-correct';
+const STD_IN_ARG = '--stdin';
+
+const DEFAULT_MESSAGE = 'Unknown Error';
+const WARNINGS = new Set(['refactor', 'convention', 'warning']);
+
+const convertOldConfig = () => {
+  const execPath = atom.config.get(OLD_EXEC_PATH_CONFIG_KEY);
+  const args = atom.config.get(OLD_ARGS_CONFIG_KEY);
+  if (!execPath && !args) { return null; }
+  atom.config.set(COMMAND_CONFIG_KEY, `${execPath || ''} ${args || ''}`.trim());
+  atom.config.set(OLD_EXEC_PATH_CONFIG_KEY, undefined);
+  return atom.config.set(OLD_ARGS_CONFIG_KEY, undefined);
+};
+
+// TODO: figure out how to best handle, also, ESLint telling me conflicting things
+// eslint-disable-next-line no-confusing-arrow
+const guard = (value, transform) => (typeof value !== 'undefined' && value !== null) ? transform(value) : undefined;
+
+const extractUrl = (parsedMessage) => {
+  const [message, url] = Array.from(parsedMessage.split(/ \((.*)\)/, 2));
+  return { message, url };
+};
+
+const formatMessage = ({ message, cop_name: copName, url }) => {
+  const formattedMessage = escapeHtml(message || DEFAULT_MESSAGE);
+  let formattedCopName = '';
+  if (copName != null && url != null) {
+    formattedCopName = ` (<a href="${escapeHtml(url)}">${escapeHtml(copName)}</a>)`;
+  } else if (copName != null) {
+    formattedCopName = ` (${escapeHtml(copName)})`;
+  }
+  return formattedMessage + formattedCopName;
+};
+
+const parseFromStd = (stdout, stderr) => {
+  let parsed;
+  try {
+    parsed = JSON.parse(stdout);
+  } catch (error) {
+    // continue regardless of error
+  }
+  if (typeof parsed !== 'object') { throw new Error(stderr || stdout); }
+  return parsed;
+};
+
+const forwardRubocopToLinter =
+  ({ message: rawMessage, location, severity, cop_name }, filePath) => {
+    const { message, url } = extractUrl(rawMessage);
+    const { line, column, length } = location || DEFAULT_LOCATION;
+    return {
+      type: WARNINGS.has(severity) ? 'Warning' : 'Error',
+      html: formatMessage({ cop_name, message, url }),
+      filePath,
+      range: [[line - 1, column - 1], [line - 1, (column + length) - 1]],
+    };
+  };
+
+const lint = function lint(editor) {
+  const filePath = editor.getPath();
+  convertOldConfig();
+  const command = atom.config.get(COMMAND_CONFIG_KEY)
+                      .split(/\s+/)
+                      .filter(i => i)
+                      .concat(DEFAULT_ARGS, STD_IN_ARG, filePath);
+  if (atom.config.get(DISABLE_CONFIG_KEY) === true) {
+    const config = helpers.find(filePath, '.rubocop.yml');
+    if (config === null) { return []; }
+  }
+  const cwd = path.dirname(helpers.find(filePath, '.'));
```

Comment:
This was the implementation in the original coffescript. What would you suggest instead? `__dirname`?